repo/blob/azure/azure_versioned_test.go

```
@@ -0,0 +1,218 @@
+	require.NoError(t, err)
+
+	defer st.DeleteBlob(ctx, blobID)
+
+	pastPIT := clock.Now().Add(-1 * time.Hour).UTC()
+	futurePIT := clock.Now().Add(1 * time.Hour).UTC()
+
+	for _, tt := range []struct {
+		testName         string
+		pointInTime      *time.Time
+		expectedBlobData string
+		expectedError    error
+	}{
+		{
+			testName:         "unset PIT",
+			pointInTime:      nil,
+			expectedBlobData: latestData,
+			expectedError:    nil,
+		},
+		{
+			testName:         "set in the future",
+			pointInTime:      &futurePIT,
+			expectedBlobData: latestData,
+			expectedError:    nil,
+		},
+		{
+			testName:         "set in the past",
+			pointInTime:      &pastPIT,
+			expectedBlobData: "",
+			expectedError:    blob.ErrBlobNotFound,
+		},
+		{
+			testName:         "original data",
+			pointInTime:      &dataTimestamps[0],
+			expectedBlobData: originalData,
+			expectedError:    nil,
+		},
+		{
+			testName:         "updated data",
+			pointInTime:      &dataTimestamps[1],
+			expectedBlobData: updatedData,
+			expectedError:    nil,
+		},
+		{
+			testName:         "latest data",
+			pointInTime:      &dataTimestamps[2],
+			expectedBlobData: latestData,
+			expectedError:    nil,
+		},
+	} {
+		fmt.Printf("Running test: %s\n", tt.testName)
+		opts.PointInTime = tt.pointInTime
+		st, err = azure.New(ctx, opts, false)
+		require.NoError(t, err)
+
+		var tmp gather.WriteBuffer
+		err = st.GetBlob(ctx, blobID, 0, -1, &tmp)
+		require.ErrorIs(t, err, tt.expectedError)
+		require.Equal(t, tt.expectedBlobData, string(tmp.ToByteSlice()))
+	}
+}
+
+func TestGetBlobVersionsWithDeletion(t *testing.T) {
+	t.Parallel()
+	testutil.ProviderTest(t)
+
+	// must be with Immutable Storage with Versioning enabled
+	container := getEnvOrSkip(t, testContainerEnv)
+	storageAccount := getEnvOrSkip(t, testStorageAccountEnv)
+	storageKey := getEnvOrSkip(t, testStorageKeyEnv)
+
+	// create container if does not exist
+	createContainer(t, container, storageAccount, storageKey)
```

Similar comments to the ones above.

Also, these tests run in parallel, so container creation is going to race.

---

Removed the `createContainer` commands since they're only in 1 other test

---

That's error prone, brittle and not future proof.
If the container does not exist and the test that does not create the container runs first, then it will fail.

---

The container will be created in `TestAzureStorage` if not already created by human intervention. The 2nd run it would pass and I guess the container creation is not frequent. Similarly the tests will fail if the StorageAccount is updated but nobody creates it.

I've re-added the `createContainer` command anyway. The race does not matter since already exists error is ignored.